metakb/transform/moa.py

```diff
@@ -0,0 +1,532 @@
+
+        :param: A list of dicts containing assertions
+        :param: A list of MOA assertion records
+        :param: A dict of MOA variant records
+        :param: A dict of MOA source records
+        :param: Keeps track of proposition and support_evidence indexes
+        :param: A dict containing assertions that have been
+            transformed to the CDM
+        """
+        for record in records:
+            gene_descriptors = self._get_gene_descriptors(
+                self._get_record(record['variant']['id'], variants))
+            descriptors = \
+                self._get_descriptors(record, variants, gene_descriptors)
+            if not descriptors:
+                continue
+            else:
+                therapy_descriptors, variation_descriptors, disease_descriptors = descriptors  # noqa: E501
+
+            propositions = \
+                self._get_tr_propositions(record, variation_descriptors,
+                                          disease_descriptors,
+                                          therapy_descriptors,
+                                          propositions_ix)
+
+            # We only want therapeutic response for now
+            if not propositions:
+                continue
+
+            documents = self._get_documents(
+                self._get_record(record['source_ids'][0], sources))
+
+            methods = self._get_method()
+            statements = self._get_statement(record, propositions,
+                                             variation_descriptors,
+                                             therapy_descriptors,
+                                             disease_descriptors,
+                                             methods, documents)
+
+            response = schemas.Response(
+                statements=statements,
+                propositions=propositions,
+                variation_descriptors=variation_descriptors,
+                gene_descriptors=gene_descriptors,
+                therapy_descriptors=therapy_descriptors,
+                disease_descriptors=disease_descriptors,
+                methods=methods,
+                documents=documents
+            ).dict(exclude_none=True)
+
+            cdm_assertions[f"moa:assertion_{record['id']}"] = response
+            responses.append(response)
+
+    def _get_descriptors(self, record, variants, gene_descriptors):
+        """Return tuple of descriptors if one exists for each type.
+
+        :param: A MOA assertion
+        :param: MOA variant records
+        :param: The corresponding gene descriptors
+        :return: Descriptors
+        """
+        therapy_descriptors = self._get_therapy_descriptors(record)
+        variation_descriptors = self._get_variation_descriptors(
+            self._get_record(record['variant']['id'], variants),
+            gene_descriptors)
+        disease_descriptors = \
+            self._get_disease_descriptors(record)
```

Comment:
I would consider doing the checks below after each descriptor, so that we aren't calling all normalizers if it's not needed.